tests/PyomoTest.py

```diff
@@ -0,0 +1,283 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Tests that TEAL can create Pyomo-style expressions when passed vardata instead of floats
+"""
+import os
+import sys
+from functools import partial
+
+import numpy as np
+import pandas as pd
+import pyomo.environ as pyo
+
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..')))
+from TEAL import CashFlows
+from TEAL import CashFlow as RunCashFlow
+
+
+# *** HELPER FUNCTIONS ***
+def build_econ_settings(cfs, life=5, dr=0.1, tax=0.21, infl=0.02184):
+  """
+    Constructs global settings for econ run
+    @ In, FIXME
+    @ Out, settings, CashFlow.GlobalSettings, settings
+  """
+  active = []
+  for comp_name, cf_list in cfs.items():
+    for cf in cf_list:
+      active.append(f'{comp_name}|{cf}')
+  print(active)
+  params = {'DiscountRate': dr,
+            'tax': tax,
+            'inflation': infl,
+            'ProjectTime': life,
+            'Indicator': {'name': ['NPV'],
+                          'active': active}
+           }
+  settings = CashFlows.GlobalSettings()
+  settings.setParams(params)
+  settings._verbosity = 0
+  return settings
+
+
+def build_generator(size, lifetime, dispatch):
+  # make the TEAL component
```

Comment:
docstrings should be added where missing

Reply:
Done